corehq/apps/hqwebapp/tests/test_default_landing_page.py

```diff
@@ -99,18 +131,19 @@ def test_formplayer_default_override(self):
     ('reports_role', MySavedReportsView.urlname),
     ('webapps_role', FormplayerMain.urlname),
     ('downloads_role', DomainDashboardView.urlname),
-    ('downloads_role', 'download_data_files', 'DATA_FILE_DOWNLOAD'),
+    ('downloads_role', 'download_data_files', privileges.DATA_FILE_DOWNLOAD),
 ], TestDefaultLandingPages)
-def test_web_user_landing_page(self, role, expected_urlname, enabled_toggle=None):
+def test_web_user_landing_page(self, role, expected_urlname, privilege=None):
     if role is not None:
         role = getattr(self, role)
     user = self._make_web_user('elodin@theuniversity.com', role=role)
     self.addCleanup(user.delete, self.domain, deleted_by=None)
     self.client.login(username=user.username, password=self.global_password)
 
-    context = flag_enabled(enabled_toggle) if enabled_toggle else contextlib.suppress()  # noop context
+    context = enable_privilege(privilege) if privilege else nullcontext()
```

Comment:
@kaapstorm I'm just curious why you did not opt to go [this](https://github.com/dimagi/commcare-hq/blob/ca65f60c58e8f7be1145859cce020338776398a3/corehq/apps/api/tests/user_resources.py#L572) route? To me it seems like less code involved for doing essentially the same thing.

Reply:
+1

Reply:
I'm not sure whether you mean (1) "why did you not decorate `test_web_user_landing_page()` with `@patch(...domain_has_privilege...)`" or (2) "why did you not use `patch()` on line 143 as a context manager". So I'll answer both:

1. This test verifies the difference in behavior when the privilege is enabled (see test case on line 134), and when it is not (see test case on line 133). If I were to decorate the test function, then the test case on line 133 will fail, because the enabled privilege will cause the user to get the wrong landing page.
2. `patch()` can be used as a context manager. I kept the test cases similar to how they had been when they were enabling a toggle instead of a privilege. But I _could_ change lines 134-136 from ...
   ```python
       # ...
       ('downloads_role', 'download_data_files', privileges.DATA_FILE_DOWNLOAD),
   ], TestDefaultLandingPages)
   def test_web_user_landing_page(self, role, expected_urlname, privilege=None):
       # ...
   ```
   ... to ...
   ```python
       # ...
       ('downloads_role', 'download_data_files', True),
   ], TestDefaultLandingPages)
   def test_web_user_landing_page(self, role, expected_urlname, enable_data_file_download=False):
       # ...
       url = reverse("domain_homepage", args=[self.domain])
       if enable_data_file_download:
           with patch('corehq.apps.users.landing_pages.domain_has_privilege', return_value=True), \
                   patch('corehq.apps.users.permissions.domain_has_privilege', return_value=True):
               response = self.client.get(url, follow=True)
       else:
           response = self.client.get(url, follow=True)
       # ...
   ```
   I think that makes the test less readable. But it does allow us to drop the `enable_privilege()` context manager on line 26. Do you think I should change it?

Reply:
I extended `privilege_enabled()` to support patching `domain_has_privilege()` in the places where we use it: 1760afc8fdd6af5b886f37d24311bc7bb48185e1